src/pint/toa.py

```diff
@@ -78,6 +78,12 @@
 JD_MJD = 2400000.5
 
 
+def _compute_hash(filename):
+    h = hashlib.sha256()
+    h.update(open(filename, "rb").read())
```

Comment:
Doesn't this mean that every .tim file will be opened and read at least twice now?  Can't we do this as part of the first read?

Reply:
They will be opened and read twice. Reading a binary file without parsing is not a very expensive operation. Trying to build the hash as they are read is a pain in the neck, because the code needs to be innterlaced with the parsing code, and because hashes operate on binary data while parsing operates on text (so one has to deal with Unicode conversions).

Reply:
I was thinking about doing it under the hood at the beginning of the parsing -- for instance reading the file as a whole as you are doing for the hashing, but then maybe using that as a StringIO object for the rest of the reading?

Reply:
There's still the annoying text/binary thing. And the cost of reading a binary file the size of our .tim files is very low. It's also going to be something of a pain to keep track of all the INCLUDEs and their hashes.

Reply:
Yup.  I don't disagree.  I don't feel too strongly and I do think that using hashes somehow is a big improvement.